### ISSUE TITLE
read(byte[], int, int) in StreamingInputStream doesn't signal the end of the stream properly

### ISSUE THREAD

**Issue by clayrat** [OpenHFT/Chronicle-Bytes#13]
I'm trying to use the `StreamingInputStream` interface and ran into an issue where reading the stream never exits. Looking at the code I see
```java
    @Override
    public int read(byte[] b, int off, int len) throws IOException {
        try {
            return in.read(b, off, len);
        } catch (IORuntimeException e) {
            throw new IOException(e);
        }
    }
```
The `in.read` method returns the length of the data read, which is always 0 or more, but the end of the stream is supposed to be signalled by `-1`. The method below, `read()` seems to handle this properly.

**Comment by peter-lawrey** [comment]
Thank you, that is a bug. Would you like to issue a pull request with fix
and/or unit test?

Regards, Peter.

**Comment by clayrat** [comment]
Ok, I'll make a PR on Monday: wrap the ```in.read``` the same ```in.readRemaining() > 0 ? ... : -1``` check (unless you have a better idea) and compose a test.

**Comment by peter-lawrey** [comment]
The read() should return how much was actually read so if this is 0 it
should -1 instead.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
